src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -629,8 +629,8 @@ private async Task ValidateViewModel(EditTenantViewModel model, bool newTenant)
                 ModelState.AddModelError(nameof(EditTenantViewModel.RequestUrlPrefix), S["Host and url prefix can not be empty at the same time."]);
             }
 
-            var allOtherShells = allSettings.Where(tenant => !string.Equals(tenant.Name, model.Name, StringComparison.OrdinalIgnoreCase));
-            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && String.Equals(tenant.RequestUrlHost, model.RequestUrlHost, StringComparison.OrdinalIgnoreCase)))
+            var allOtherShells = allSettings.Where(t => !string.Equals(t.Name, model.Name, StringComparison.OrdinalIgnoreCase));
+            if (allOtherShells.Any(t => String.Equals(t.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && t.RequestUrlHost.Contains(model.RequestUrlHost, StringComparison.OrdinalIgnoreCase)))
```

Comment:
The same logic is there in `ApiController`. At this point, I'd factor it out to a common place instead of maintaining this in both places. This would also make testing easier.

Reply:
This is common in OC if I'm not wrong in this case where do you suggest the common place?

Reply:
I mean, factor out to at least a helper, or a validation service, or something that both controllers use. Otherwise, others will make the same mistake of updating it only in one place.

Reply:
If there are many place we can found such thing in Controllers & ApiController, I think we need a separate PR so the OC members can agree on the suggested solutions

Thanks

Reply:
For other places yes, but here I'm talking only about this specific one. You can circumvent this validation if you use the API, which shouldn't be the case.